solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java

```diff
@@ -54,11 +54,11 @@ public void write(OutputStream out, SolrQueryRequest request, SolrQueryResponse
       throws IOException {
     NamedList<Object> prometheusRegistries =
         (NamedList<Object>) response.getValues().get("metrics");
-    Iterator<Map.Entry<String, Object>> it = prometheusRegistries.iterator();
     var prometheusTextFormatWriter = new PrometheusTextFormatWriter(false);
-    while (it.hasNext()) {
-      Map.Entry<String, Object> entry = it.next();
-      SolrPrometheusCoreExporter prometheusExporter = (SolrPrometheusCoreExporter) entry.getValue();
+    for (Iterator<Map.Entry<String, Object>> iter = prometheusRegistries.iterator();
```

Comment:
This isn't what I meant.  If you use IntelliJ, it should give you a hint -- "'for' loop can be replaced with enhanced 'for' ".  Do that and I think you'll appreciate the results.

Reply:
Sorry was never they best at Java. The enhanced for from the hint is a lot better